### PR TITLE
Fix bug in Lzo.decompressSafe

### PR DESCRIPTION
**Pull request by gavin-norman-sociomantic**

### CODE REVIEW

File: src/ocean/io/compress/Lzo.d [hunk 1]
```diff
@@ -243,6 +243,23 @@ class Lzo
     }
 }
 
+/// Simple compress / decompress example.
+unittest
+{
+    auto lzo = new Lzo;
+    auto original = "compress this!".dup;
```
Comment (david-eckardt-sociomantic) [inline src/ocean/io/compress/Lzo.d]: The `.dup` should not be needed.

File: src/ocean/io/compress/Lzo.d [hunk 2]
```diff
@@ -243,6 +243,23 @@ class Lzo
...
+
+    // Compress.
+    auto compressed = new void[lzo.maxCompressedLength(original.length)];
```
Comment (david-eckardt-sociomantic) [inline src/ocean/io/compress/Lzo.d]: Not very important, but without the `.dup` above `compressed` and `uncompressed` can be static `ubyte[n]` arrays.

File: src/ocean/io/compress/Lzo.d [hunk 3]
```diff
@@ -243,6 +243,23 @@ class Lzo
...
+    compressed.length = lzo.compress(original, compressed);
+
+    // Decompress (with overrun safety checks).
+    auto uncompressed = new void[original.length];
+    lzo.decompressSafe(compressed, uncompressed);
```
Comment (david-eckardt-sociomantic) [inline src/ocean/io/compress/Lzo.d]: Test that it returns `uncompressed.length`.

File: src/ocean/io/compress/Lzo.d (hunk 1 above)
Comment (gavin-norman-sociomantic) [reply src/ocean/io/compress/Lzo.d]: Right, true, it doesn't need to be mutable.

File: src/ocean/io/compress/Lzo.d (hunk 2 above)
Comment (gavin-norman-sociomantic) [reply src/ocean/io/compress/Lzo.d]: I think that's less intuitive as a usage example.

File: src/ocean/io/compress/Lzo.d (hunk 3 above)
Comment (gavin-norman-sociomantic) [reply src/ocean/io/compress/Lzo.d]: As this is primarily intended to be a usage example (documented unittest), I'm not sure about that. In the example given, the return value is irrelevant, as we've already ensured that the decompress destination buffer is exactly the right size.

File: src/ocean/io/compress/Lzo.d (hunk 1 above)
Comment (gavin-norman-sociomantic) [reply src/ocean/io/compress/Lzo.d]: Updated.

File: src/ocean/io/compress/Lzo.d (hunk 3 above)
Comment (david-eckardt-sociomantic) [reply src/ocean/io/compress/Lzo.d]: Maybe add a comment, otherwise it may look as if the return value is irrelevant.

**Comment by gavin-norman-sociomantic** [comment]
Updated: reworked usage example to demonstrate the common case of receiving a compressed buffer over the network.

File: src/ocean/io/compress/Lzo.d [hunk 4]
```diff
@@ -243,6 +243,41 @@ class Lzo
...
+    // LZO instance. (Usually a single one can be shared globally.)
+    auto lzo = new Lzo;
+
+    // Typical function to decompress a buffer that was received over the
+    // network. (i.e. where the buffer and the expected length of the
+    // decompressed data are both received from an external source.)
+    void decompress ( in void[] src, size_t expected_decompressed_length,
+        ref void[] dst )
+    {
+        dst.length = expected_decompressed_length;
```
Comment (nemanja-boric-sociomantic) [inline src/ocean/io/compress/Lzo.d:258]: _maybe_ `enableStomping` ?

**Review by nemanja-boric-sociomantic — APPROVED** [review]

**Review by david-eckardt-sociomantic — APPROVED** [review]

File: src/ocean/io/compress/Lzo.d (hunk 4 above)
Comment (gavin-norman-sociomantic) [reply src/ocean/io/compress/Lzo.d:258]: :+1: Updated.

**Pull request merged**